### PR TITLE
chore: ☂️ Migrate BundleHermesCTaskTest.kt to AssertJ

### PR DESCRIPTION
**Pull request by omer-ayhan**
## Summary:
Issue: https://github.com/facebook/react-native/issues/45596

## Changelog:
[INTERNAL] [CHANGED] - Migrate `BundleHermesCTaskTest.kt` to AssertJ testing library

## Test Plan:
Run `./gradlew -p packages/gradle-plugin test`

### CODE REVIEW

**Comment by omer-ayhan** [comment]
hello @cortinico 👋 this PR is ready for the review🚀

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 1]
```diff
@@ -220,31 +213,31 @@ class BundleHermesCTaskTest {
...
-    assertEquals("--sourcemap-output", bundleCommand[16])
-    assertEquals(sourceMapFile.absolutePath, bundleCommand[17])
-    assertEquals("--config", bundleCommand[18])
-    assertEquals(bundleConfig.absolutePath, bundleCommand[19])
-    assertEquals("--minify", bundleCommand[20])
-    assertEquals("true", bundleCommand[21])
-    assertEquals("--read-global-cache", bundleCommand[22])
-    assertEquals("--verbose", bundleCommand[23])
-    assertEquals(24, bundleCommand.size)
+    assertThat(bundleCommand[0]).isEqualTo("node")
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:216]: Can you convert this to a `containsExactly` assertion?

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 2]
```diff
@@ -272,33 +265,33 @@ class BundleHermesCTaskTest {
...
-    assertEquals("--sourcemap-output", bundleCommand[18])
-    assertEquals(sourceMapFile.relativeTo(tempFolder.root).path, bundleCommand[19])
-    assertEquals("--config", bundleCommand[20])
-    assertEquals(bundleConfig.relativeTo(tempFolder.root).path, bundleCommand[21])
-    assertEquals("--minify", bundleCommand[22])
-    assertEquals("true", bundleCommand[23])
-    assertEquals("--read-global-cache", bundleCommand[24])
-    assertEquals("--verbose", bundleCommand[25])
-    assertEquals(26, bundleCommand.size)
+    assertThat(bundleCommand[0]).isEqualTo("cmd")
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:268]: Can you convert this to a `containsExactly` assertion?

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 3]
```diff
@@ -339,14 +332,14 @@ class BundleHermesCTaskTest {
...
 
-    assertEquals(customHermesc, hermesCommand[0])
-    assertEquals("-emit-binary", hermesCommand[1])
-    assertEquals("-max-diagnostic-width=80", hermesCommand[2])
-    assertEquals("-out", hermesCommand[3])
-    assertEquals(bytecodeFile.absolutePath, hermesCommand[4])
-    assertEquals(bundleFile.absolutePath, hermesCommand[5])
-    assertEquals("my-custom-hermes-flag", hermesCommand[6])
-    assertEquals(7, hermesCommand.size)
+    assertThat(hermesCommand[0]).isEqualTo(customHermesc)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:335]: same as above

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 4]
```diff
@@ -363,16 +356,16 @@ class BundleHermesCTaskTest {
...
-    assertEquals("/c", hermesCommand[1])
-    assertEquals(customHermesc, hermesCommand[2])
-    assertEquals("-emit-binary", hermesCommand[3])
-    assertEquals("-max-diagnostic-width=80", hermesCommand[4])
-    assertEquals("-out", hermesCommand[5])
-    assertEquals(bytecodeFile.relativeTo(tempFolder.root).path, hermesCommand[6])
-    assertEquals(bundleFile.relativeTo(tempFolder.root).path, hermesCommand[7])
-    assertEquals("my-custom-hermes-flag", hermesCommand[8])
-    assertEquals(9, hermesCommand.size)
+    assertThat(hermesCommand[0]).isEqualTo("cmd")
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:359]: same as above

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 5]
```diff
@@ -391,15 +384,15 @@ class BundleHermesCTaskTest {
...
-    assertEquals("node", composeSourcemapCommand[0])
-    assertEquals("arg1", composeSourcemapCommand[1])
-    assertEquals("arg2", composeSourcemapCommand[2])
-    assertEquals(composeSourceMapsFile.absolutePath, composeSourcemapCommand[3])
-    assertEquals(packagerMap.absolutePath, composeSourcemapCommand[4])
-    assertEquals(compilerMap.absolutePath, composeSourcemapCommand[5])
-    assertEquals("-o", composeSourcemapCommand[6])
-    assertEquals(outputMap.absolutePath, composeSourcemapCommand[7])
-    assertEquals(8, composeSourcemapCommand.size)
+    assertThat(composeSourcemapCommand[0]).isEqualTo("node")
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:387]: same as above

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 6]
```diff
@@ -419,16 +412,16 @@ class BundleHermesCTaskTest {
...
-    assertEquals("node", composeSourcemapCommand[2])
-    assertEquals("arg1", composeSourcemapCommand[3])
-    assertEquals("arg2", composeSourcemapCommand[4])
-    assertEquals(composeSourceMapsFile.relativeTo(tempFolder.root).path, composeSourcemapCommand[5])
-    assertEquals(packagerMap.relativeTo(tempFolder.root).path, composeSourcemapCommand[6])
-    assertEquals(compilerMap.relativeTo(tempFolder.root).path, composeSourcemapCommand[7])
-    assertEquals("-o", composeSourcemapCommand[8])
-    assertEquals(outputMap.relativeTo(tempFolder.root).path, composeSourcemapCommand[9])
-    assertEquals(10, composeSourcemapCommand.size)
+    assertThat(composeSourcemapCommand[0]).isEqualTo("cmd")
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:415]: same as above

**Review by cortinico — CHANGES_REQUESTED** [review]

**Comment by omer-ayhan** [comment]
changes are done @cortinico

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 7]
```diff
@@ -339,14 +332,14 @@ class BundleHermesCTaskTest {
...
 
-    assertEquals(customHermesc, hermesCommand[0])
-    assertEquals("-emit-binary", hermesCommand[1])
-    assertEquals("-max-diagnostic-width=80", hermesCommand[2])
-    assertEquals("-out", hermesCommand[3])
-    assertEquals(bytecodeFile.absolutePath, hermesCommand[4])
-    assertEquals(bundleFile.absolutePath, hermesCommand[5])
-    assertEquals("my-custom-hermes-flag", hermesCommand[6])
-    assertEquals(7, hermesCommand.size)
+    assertThat(hermesCommand[0]).containsExactly(customHermesc)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:335]: @omer-ayhan Here with `containsExactly` I meant to assert all the parameters.

Like:

```
assertThat(hermesCommand).containsExactly(
  customHermesc,
  "-emit-binary",
  "-max-diagnostic-width=80"
...
)
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 8]
```diff
@@ -363,16 +356,16 @@ class BundleHermesCTaskTest {
...
-    assertEquals("/c", hermesCommand[1])
-    assertEquals(customHermesc, hermesCommand[2])
-    assertEquals("-emit-binary", hermesCommand[3])
-    assertEquals("-max-diagnostic-width=80", hermesCommand[4])
-    assertEquals("-out", hermesCommand[5])
-    assertEquals(bytecodeFile.relativeTo(tempFolder.root).path, hermesCommand[6])
-    assertEquals(bundleFile.relativeTo(tempFolder.root).path, hermesCommand[7])
-    assertEquals("my-custom-hermes-flag", hermesCommand[8])
-    assertEquals(9, hermesCommand.size)
+    assertThat(hermesCommand[0]).containsExactly("cmd")
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:359]: same as above

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 9]
```diff
@@ -391,15 +384,15 @@ class BundleHermesCTaskTest {
...
-    assertEquals("node", composeSourcemapCommand[0])
-    assertEquals("arg1", composeSourcemapCommand[1])
-    assertEquals("arg2", composeSourcemapCommand[2])
-    assertEquals(composeSourceMapsFile.absolutePath, composeSourcemapCommand[3])
-    assertEquals(packagerMap.absolutePath, composeSourcemapCommand[4])
-    assertEquals(compilerMap.absolutePath, composeSourcemapCommand[5])
-    assertEquals("-o", composeSourcemapCommand[6])
-    assertEquals(outputMap.absolutePath, composeSourcemapCommand[7])
-    assertEquals(8, composeSourcemapCommand.size)
+    assertThat(composeSourcemapCommand[0]).containsExactly("node")
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:387]: same as above

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 10]
```diff
@@ -419,16 +412,16 @@ class BundleHermesCTaskTest {
...
-    assertEquals("node", composeSourcemapCommand[2])
-    assertEquals("arg1", composeSourcemapCommand[3])
-    assertEquals("arg2", composeSourcemapCommand[4])
-    assertEquals(composeSourceMapsFile.relativeTo(tempFolder.root).path, composeSourcemapCommand[5])
-    assertEquals(packagerMap.relativeTo(tempFolder.root).path, composeSourcemapCommand[6])
-    assertEquals(compilerMap.relativeTo(tempFolder.root).path, composeSourcemapCommand[7])
-    assertEquals("-o", composeSourcemapCommand[8])
-    assertEquals(outputMap.relativeTo(tempFolder.root).path, composeSourcemapCommand[9])
-    assertEquals(10, composeSourcemapCommand.size)
+    assertThat(composeSourcemapCommand[0]).containsExactly("cmd")
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:415]: same as above

**Review by cortinico — CHANGES_REQUESTED** [review]

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt (hunk 7 above)
Comment (omer-ayhan) [reply packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:335]: sorry my bad I'll do it as soon as I can @cortinico

**Comment by omer-ayhan** [comment]
solved the issues @cortinico

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 11]
```diff
@@ -220,31 +213,33 @@ class BundleHermesCTaskTest {
...
+      "--sourcemap-output",
+      sourceMapFile.absolutePath,
+      "--config",
+      bundleConfig.absolutePath,
+      "--minify",
+      "true",
+      "--read-global-cache",
+      "--verbose"
+    )
+    assertThat(bundleCommand.size).isEqualTo(24)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:242]:
```suggestion
    assertThat(bundleCommand).containsExactly(
      "node",
      "arg1",
      "arg2",
      cliFile.absolutePath,
      "bundle",
      "--platform",
      "android",
      "--dev",
      "true",
      "--reset-cache",
      "--entry-file",
      entryFile.absolutePath,
      "--bundle-output",
      bundleFile.absolutePath,
      "--assets-dest",
      resourcesDir.absolutePath,
      "--sourcemap-output",
      sourceMapFile.absolutePath,
      "--config",
      bundleConfig.absolutePath,
      "--minify",
      "true",
      "--read-global-cache",
      "--verbose"
    )
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 12]
```diff
@@ -272,33 +267,35 @@ class BundleHermesCTaskTest {
...
+      "--sourcemap-output",
+      sourceMapFile.relativeTo(tempFolder.root).path,
+      "--config",
+      bundleConfig.relativeTo(tempFolder.root).path,
+      "--minify",
+      "true",
+      "--read-global-cache",
+      "--verbose"
+    )
+    assertThat(bundleCommand.size).isEqualTo(26)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:298]:
```suggestion
    assertThat(bundleCommand).containsExactly(
      "cmd",
      "/c",
      "node",
      "arg1",
      "arg2",
      cliFile.relativeTo(tempFolder.root).path,
      "bundle",
      "--platform",
      "android",
      "--dev",
      "true",
      "--reset-cache",
      "--entry-file",
      entryFile.relativeTo(tempFolder.root).path,
      "--bundle-output",
      bundleFile.relativeTo(tempFolder.root).path,
      "--assets-dest",
      resourcesDir.relativeTo(tempFolder.root).path,
      "--sourcemap-output",
      sourceMapFile.relativeTo(tempFolder.root).path,
      "--config",
      bundleConfig.relativeTo(tempFolder.root).path,
      "--minify",
      "true",
      "--read-global-cache",
      "--verbose"
    )
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 13]
```diff
@@ -339,14 +336,16 @@ class BundleHermesCTaskTest {
...
+    assertThat(hermesCommand).containsExactly(
+      customHermesc,
+      "-emit-binary",
+      "-max-diagnostic-width=80",
+      "-out",
+      bytecodeFile.absolutePath,
+      bundleFile.absolutePath,
+      "my-custom-hermes-flag"
+      )
+    assertThat(hermesCommand.size).isEqualTo(7)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:348]:
```suggestion
    assertThat(hermesCommand).containsExactly(
      customHermesc,
      "-emit-binary",
      "-max-diagnostic-width=80",
      "-out",
      bytecodeFile.absolutePath,
      bundleFile.absolutePath,
      "my-custom-hermes-flag"
      )
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 14]
```diff
@@ -363,16 +362,18 @@ class BundleHermesCTaskTest {
...
+      "/c",
+      customHermesc,
+      "-emit-binary",
+      "-max-diagnostic-width=80",
+      "-out",
+      bytecodeFile.relativeTo(tempFolder.root).path,
+      bundleFile.relativeTo(tempFolder.root).path,
+      "my-custom-hermes-flag"
+    )
+    assertThat(hermesCommand.size).isEqualTo(9)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:376]:
```suggestion
    assertThat(hermesCommand).containsExactly(
      "cmd",
      "/c",
      customHermesc,
      "-emit-binary",
      "-max-diagnostic-width=80",
      "-out",
      bytecodeFile.relativeTo(tempFolder.root).path,
      bundleFile.relativeTo(tempFolder.root).path,
      "my-custom-hermes-flag"
    )
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 15]
```diff
@@ -391,15 +392,17 @@ class BundleHermesCTaskTest {
...
+      "node",
+      "arg1",
+      "arg2",
+      composeSourceMapsFile.absolutePath,
+      packagerMap.absolutePath,
+      compilerMap.absolutePath,
+      "-o",
+      outputMap.absolutePath
+    )
+    assertThat(composeSourcemapCommand.size).isEqualTo(8)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:405]:
```suggestion
    assertThat(composeSourcemapCommand).containsExactly(
      "node",
      "arg1",
      "arg2",
      composeSourceMapsFile.absolutePath,
      packagerMap.absolutePath,
      compilerMap.absolutePath,
      "-o",
      outputMap.absolutePath
    )
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt [hunk 16]
```diff
@@ -419,16 +422,18 @@ class BundleHermesCTaskTest {
...
+      "node",
+      "arg1",
+      "arg2",
+      composeSourceMapsFile.relativeTo(tempFolder.root).path,
+      packagerMap.relativeTo(tempFolder.root).path,
+      compilerMap.relativeTo(tempFolder.root).path,
+      "-o",
+      outputMap.relativeTo(tempFolder.root).path
+    )
+    assertThat(composeSourcemapCommand.size).isEqualTo(10)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/BundleHermesCTaskTest.kt:437]:
```suggestion
    assertThat(composeSourcemapCommand).containsExactly(
      "cmd",
      "/c",
      "node",
      "arg1",
      "arg2",
      composeSourceMapsFile.relativeTo(tempFolder.root).path,
      packagerMap.relativeTo(tempFolder.root).path,
      compilerMap.relativeTo(tempFolder.root).path,
      "-o",
      outputMap.relativeTo(tempFolder.root).path
    )
```

**Review by cortinico — APPROVED** [review]

**Comment by omer-ayhan** [comment]
I didn't know you wanted to delete `isEqualTo` as well @cortinico

**Comment by cortinico** [comment]
> I didn't know you wanted to delete isEqualTo as well @cortinico

@omer-ayhan the `isEqualTo` assertion on the size is unnecessary because of the `containsExactly` assertion just above.

**Pull request closed without merging**